.github/api_guidelines.md

```diff
@@ -8,10 +8,130 @@ in slack channel [#forum-acs-api-design](https://redhat-internal.slack.com/archi
 ## Table of Contents
 
 - [Naming Guidelines](#naming-guidelines)
+  - [Decoupling](decoupling)
   - [gRPC Service Name](#grpc-service-name)
   - [gRPC Method Name](#grpc-method-name)
   - [gRPC Message Name](#grpc-message-name)
 
+## Decoupling
+
+While certain data may exist in the database for internal purposes or other functionalities, it is important to 
+exercise caution when deciding what data to make accessible through the API. The rule of thumb to follow is to **not 
+expose data structures defined for internal purposes at APIs**. Define a data structure for sole purpose of 
+API use, generally in the same package were the API is defined. This applies to all APIs including the 
+inter-component APIs.
+- All services defined in the package [/proto/api/v2](https://github.com/stackrox/stackrox/blob/master/proto/api/v2)
+**must not** import from the following packages:
+  - [/proto/storage](https://github.com/stackrox/stackrox/blob/master/proto/storage) 
+  - [/proto/internalapi](https://github.com/stackrox/stackrox/blob/master/proto/internalapi) 
+  - [/proto/api/v1](https://github.com/stackrox/stackrox/blob/master/proto/api/v1)
+- All new services and methods defined in the package [/proto/api/v1](https://github.com/stackrox/stackrox/blob/master/proto/api/v1) 
+**must not** import from the following packages:
+  - [/proto/storage](https://github.com/stackrox/stackrox/blob/master/proto/storage)
+  - [/proto/internalapi](https://github.com/stackrox/stackrox/blob/master/proto/internalapi)
+  - [/proto/api/v2](https://github.com/stackrox/stackrox/blob/master/proto/api/v2)
+- All new messages defined in the package [/proto/internalapi](https://github.com/stackrox/stackrox/blob/master/proto/internalapi)
+  **must not** import from the following packages:
+  - [/proto/storage](https://github.com/stackrox/stackrox/blob/master/proto/storage)
+  - [/proto/api/v1](https://github.com/stackrox/stackrox/blob/master/proto/api/v1)
+  - [/proto/api/v2](https://github.com/stackrox/stackrox/blob/master/proto/api/v2)
+- All services of the type `APIServiceWithCustomRoutes` **must not** import from 
+[/proto/storage](https://github.com/stackrox/stackrox/blob/master/proto/storage) and 
+[/proto/internalapi](https://github.com/stackrox/stackrox/blob/master/proto/internalapi) packages, at the minimum.
+- Structs used by APIs, such as those defined in [/proto/api/](https://github.com/stackrox/stackrox/blob/master/proto/api/), 
+[/proto/internalapi](https://github.com/stackrox/stackrox/blob/master/proto/internalapi), **should not** be 
```

Comment:
Are there cases where these are written to the DB today?  Curious why you've used **should not** here instead of **must not**.

Reply:
I was unsure about it as well. Grepping through the code base to check if any stores use API protos and concluded that we do not. Made the guideline stricter.